Only accept control transactions over the control transaction topic

hrxi opened the pull request:
Do this by using a wrapper that only accepts control transactions.

Fixes #2614.

hrxi committed:
Only accept control transactions over the control transaction topic

Do this by using a wrapper that only accepts control transactions.

Fixes #2614.

Eligioo commented on primitives/transaction/src/control_transaction.rs:
```diff
@@ -0,0 +1,65 @@
+    type Error = ControlTransactionError;
+
+    fn try_from(tx: Transaction) -> Result<ControlTransaction, ControlTransactionError> {
+        if tx.sender_type != AccountType::Staking && tx.recipient_type != AccountType::Staking {
```
Should we also check if the value is 0 here?

hrxi replied:
We discussed this shortly, and the answer was yes. However, since that is a breaking change, it cannot go into the next (compatible) testnet release.

sisou replied:
Signaling transactions also have `TransactionFlags::SIGNALING` set, but AFAIK the combination of that flag and zero value is already checked in the transaction's internal validity and/or parser.

The parsed `IncomingStakingTransactionData` has a `.is_signaling()` method:
https://github.com/nimiq/core-rs-albatross/blob/21d91b3187e234c3319d9e3d774512fb04eb3c35/primitives/transaction/src/account/staking_contract/structs.rs#L90

dependabot[bot] committed:
build(deps): bump docker/build-push-action from 5.4.0 to 6.0.0

Bumps [docker/build-push-action](https://github.com/docker/build-push-action) from 5.4.0 to 6.0.0.
- [Release notes](https://github.com/docker/build-push-action/releases)
- [Commits](https://github.com/docker/build-push-action/compare/ca052bb54ab0790a636c9b5f226502c73d547a25...c382f710d39a5bb4e430307530a720f50c2d3318)

---
updated-dependencies:
- dependency-name: docker/build-push-action
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

nibhar committed:
continue polling Tendermint when old state update is received.

hrxi committed:
Add some tests around control transactions

viquezclaudio committed:
Fix clippy warning

nibhar committed:
Tendermint: Choose better future contribution to verify than just random

Don't verify contributions whose contributors are a subset of the already verified contributors.

dependabot[bot] committed:
build(deps): bump derive_more in the minor-and-patch-updates group

Bumps the minor-and-patch-updates group with 1 update: [derive_more](https://github.com/JelteF/derive_more).


Updates `derive_more` from 0.99.17 to 0.99.18
- [Release notes](https://github.com/JelteF/derive_more/releases)
- [Changelog](https://github.com/JelteF/derive_more/blob/v0.99.18/CHANGELOG.md)
- [Commits](https://github.com/JelteF/derive_more/compare/v0.99.17...v0.99.18)

---
updated-dependencies:
- dependency-name: derive_more
  dependency-type: direct:production
  update-type: version-update:semver-patch
  dependency-group: minor-and-patch-updates
...

Signed-off-by: dependabot[bot] <support@github.com>
